include/seqan/score/score_matrix_dyn.h

```diff
@@ -0,0 +1,239 @@
+ * @brief Blosum62, see also @link Blosum62 @endlink
+ *
+ * @val AminoAcidScoreMatrixID AminoAcidScoreMatrixID::BLOSUM80
+ * @brief Blosum80, see also @link Blosum80 @endlink
+ *
+ * @val AminoAcidScoreMatrixID AminoAcidScoreMatrixID::PAM40
+ * @brief Pam40, see also @link Pam40 @endlink
+ *
+ * @val AminoAcidScoreMatrixID AminoAcidScoreMatrixID::PAM120
+ * @brief Pam120, see also @link Pam120 @endlink
+ *
+ * @val AminoAcidScoreMatrixID AminoAcidScoreMatrixID::PAM200
+ * @brief Pam200, see also @link Pam200 @endlink
+ *
+ * @val AminoAcidScoreMatrixID AminoAcidScoreMatrixID::PAM250
+ * @brief Pam250, see also @link Pam250 @endlink
+ *
+ * @val AminoAcidScoreMatrixID AminoAcidScoreMatrixID::VTML200
+ * @brief Vtml200, see also @link Vtml200 @endlink
+ */
+
+enum class AminoAcidScoreMatrixID : uint8_t
+{
+    BLOSUM30,
+    BLOSUM45,
+    BLOSUM62,
+    BLOSUM80,
+    PAM40,
+    PAM120,
+    PAM200,
+    PAM250,
+    VTML200
+};
+
+// ----------------------------------------------------------------------------
+// Class SelectableAminoAcidMatrix
+// ----------------------------------------------------------------------------
+
+/*!
+ * @typedef SelectableAminoAcidMatrix
+ * @headerfile <seqan/score.h>
+ * @brief An AminoAcid score matrix that can be "specialized" at run-time
+ *
+ * @signature using SelectableAminoAcidMatrix =  Score<int, ScoreMatrix<AminoAcid, ScoreSpecSelectable> >;
+ */
+
+template <typename TValue>
+class Score<TValue, ScoreMatrix<AminoAcid, ScoreSpecSelectable> >
+ : public Score<TValue, ScoreMatrix<AminoAcid, Default> >
+{
+public:
+    AminoAcidScoreMatrixID _ident = static_cast<AminoAcidScoreMatrixID>(0u);
+};
+
+using SelectableAminoAcidMatrix =  Score<int, ScoreMatrix<AminoAcid, ScoreSpecSelectable> >;
+
+// ============================================================================
+// Functions
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// helper functions
+// ----------------------------------------------------------------------------
+
+namespace impl
+{
+namespace score
+{
+
+template <typename TCurTag,
+          typename TRunnable>
+inline void
+matrixTagDispatch(TagList<TCurTag, void> const &,
+                  AminoAcidScoreMatrixID const m,
+                  TRunnable const & runnable)
+{
+    if (LENGTH<MatrixTags>::VALUE == static_cast<uint8_t>(m))
+        runnable(TCurTag());
+    else
+        SEQAN_FAIL("ERROR: Recursing the ScoreMatrixTags failed, please report this as a BUG!");
+}
+
+template <typename TCurTag,
+          typename TRestList,
+          typename TRunnable>
+inline void
+matrixTagDispatch(TagList<TCurTag, TRestList> const &,
+                  AminoAcidScoreMatrixID const m,
+                  TRunnable const & runnable)
+{
+    using TCurList = TagList<TCurTag, TRestList>;
+    if (LENGTH<MatrixTags>::VALUE  - LENGTH<TCurList>::VALUE == static_cast<uint8_t>(m))
```

Comment:
Please, don't use a new enum class.
You can use the Find metafunction and the TagSelector class in fundamental_tags.h:  https://github.com/seqan/seqan/blob/develop/include/seqan/basic/fundamental_tags.h#L481
In this case you don't need to worry about the correct order of the tags in the TagList and the enum, which is very evil.
```cpp
if (value(selector) == Find<TagList, ScoreSpecPam120>::VALUE)
```

Then you just pass the ScoreSpec tag instead of a new id, which is not necessary.